tritium-registry/src/main/java/com/palantir/tritium/metrics/registry/TaggedMetricRegistry.java

```diff
@@ -111,7 +111,7 @@ default void registerWithReplacement(MetricName metricName, Gauge<?> gauge) {
         }
         remove(metricName).ifPresent(removed -> LoggerFactory.getLogger(getClass())
                 .debug("Removed previously registered gauge {}", SafeArg.of("metricName", metricName)));
-        gauge(metricName, gauge);
+        registerWithReplacement(metricName, gauge);
```

Comment:
@carterkozak @iamdanfox the robots have definitely recursed here as this would infinite loop. What's the best way to suppress this automated refactor?

```suggestion
        gauge(metricName, gauge);
```

Reply:
Alternatively we could add a check that `gauge(metricName, gauge);` has registered `gauge` since there's a race condition which can allow another invocation to race the current call.

The error-prone check only applies to calls where the return value is ignored.